modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/PublicationChannelToWorkspace.java

```diff
@@ -0,0 +1,153 @@
+
+    WorkflowOperationInstance currentOperation = workflowInstance.getCurrentOperation();
+
+    ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance, Configuration.many,
+        Configuration.many, Configuration.many, Configuration.many);
+    List<MediaPackageElementFlavor> configuredSourceFlavors = tagsAndFlavors.getSrcFlavors();
+    List<String> configuredSourceTags = tagsAndFlavors.getSrcTags();
+    List <String> configuredTargetTags = tagsAndFlavors.getTargetTags();
+    String publicationChannel = StringUtils
+        .trimToEmpty(currentOperation.getConfiguration(OPT_SOURCE_PUBLICATION_CHANNEL));
+    String configuredTargetTagsAsString =  String.join(",", configuredTargetTags);
+
+    if (publicationChannel.isEmpty()) {
+      logger.error("No source publication-channel set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+    if (configuredSourceFlavors.isEmpty()) {
+      logger.error("No source flavor set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+    if (configuredSourceTags.isEmpty()) {
+      logger.error("No source Tag set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+
+    Optional<Publication> publication = Arrays.stream(mediaPackage.getPublications())
+        .filter(channel -> channel.getChannel().equals(publicationChannel)).findFirst();
+    if (publication.get().getTracks().length >= 0) {
+
+      Collection<MediaPackageElement> tracks =  new ArrayList<MediaPackageElement>();
+
+      Arrays.stream(publication.get().getTracks())
+          .filter(element -> element.containsTag(configuredSourceTags))
+          .filter(Objects::nonNull)
+          .forEach(element -> tracks.add(element));
+      Arrays.stream(publication.get().getTracks())
+          .filter(element -> configuredSourceFlavors.contains(element.getFlavor()))
+          .filter(Objects::nonNull)
+          .forEach(element -> tracks.add(element));
+
+      tracks.stream().forEach(element -> element.addTag(configuredTargetTagsAsString));
+      tracks.stream().forEach(mediaPackageElement -> mediaPackage.add(mediaPackageElement));
+    }
+
+    if (publication.get().getAttachments().length >= 0) {
+      Collection<MediaPackageElement> attachments = new ArrayList<MediaPackageElement>();
+      Arrays.stream(publication.get().getAttachments())
+          .filter(element -> element.containsTag(configuredSourceTags))
+          .filter(Objects::nonNull)
+          .forEach(element -> attachments.add(element));
+      Arrays.stream(publication.get().getAttachments())
+          .filter(element -> configuredSourceFlavors.contains(element.getFlavor()))
+          .filter(Objects::nonNull)
+          .forEach(element -> attachments.add(element));
+
+      attachments.stream().forEach(element -> element.addTag(configuredTargetTagsAsString));
```

Comment:
If you configures `a,b` as tags, this will not add `a` and `b` as tags, but rather `a,b`.
That is not intended, I guess.